plutus-use-cases/test/Spec/Future.hs

```diff
@@ -42,38 +42,40 @@ tests =
         $ void F.setupTokensTrace
 
     , checkPredicate "can initialise and obtain tokens"
-        (walletFundsChange w1 (scale (-1) (F.initialMargin theFuture) <> F.tokenFor Short F.testAccounts)
-        .&&. walletFundsChange w2 (scale (-1) (F.initialMargin theFuture) <> F.tokenFor Long F.testAccounts))
-        (void (initContract >> joinFuture))
+        (walletFundsChange w1 (scale (-1) (F.initialMargin (theFuture startTime)) <> F.tokenFor Short F.testAccounts)
+        .&&. walletFundsChange w2 (scale (-1) (F.initialMargin (theFuture startTime)) <> F.tokenFor Long F.testAccounts))
+        (void (initContract startTime >> joinFuture startTime))
 
     , checkPredicate "can increase margin"
         (assertAccountBalance (ftoShort F.testAccounts) (== Ada.lovelaceValueOf 1936)
         .&&. assertAccountBalance (ftoLong F.testAccounts) (== Ada.lovelaceValueOf 2410))
-        increaseMarginTrace
+        (increaseMarginTrace startTime)
 
     , checkPredicate "can settle early"
         (assertAccountBalance (ftoShort F.testAccounts) (== Ada.lovelaceValueOf 0)
         .&&. assertAccountBalance (ftoLong F.testAccounts) (== Ada.lovelaceValueOf 4246))
-        settleEarlyTrace
+        (settleEarlyTrace startTime)
 
      , checkPredicate "can pay out"
         (assertAccountBalance (ftoShort F.testAccounts) (== Ada.lovelaceValueOf 1936)
         .&&. assertAccountBalance (ftoLong F.testAccounts) (== Ada.lovelaceValueOf 2310))
-        payOutTrace
+        (payOutTrace startTime)
 
     , goldenPir "test/Spec/future.pir" $$(PlutusTx.compile [|| F.futureStateMachine ||])
 
     , HUnit.testCaseSteps "script size is reasonable" $ \step ->
-        reasonable' step (F.validator theFuture F.testAccounts) 63000
-
+        reasonable' step (F.validator (theFuture startTime) F.testAccounts) 63000
     ]
 
-setup :: FutureSetup
-setup =
+    where
+        startTime = TimeSlot.scSlotZeroTime def
+
+setup :: POSIXTime -> FutureSetup
```

Comment:
Same here (I won't comment on all the other ones, but it should be fine if we make `startTime` a top-level definition, and use it everywhere in the module